modules/dnn/test/test_layers_1d.cpp

```diff
@@ -239,10 +239,57 @@ TEST_P(Layer_Elemwise_1d_Test, Accuracy) {
         CV_Error(Error::StsAssert, "Provided operation: " + operation + " is not supported. Please check the test instantiation.");
     }
 }
-
 INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*input blob shape*/    Values(0, 1, 2, 3),
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+typedef testing::TestWithParam<tuple<tuple<int, std::vector<int>>, int>> Layer_Softmax_Test;
+TEST_P(Layer_Softmax_Test, Accuracy) {
+
+    auto tup = get<0>(GetParam());
+    int axis = get<1>(GetParam());
+    int dims = get<0>(tup);
+    std::vector<int> input_shape = get<1>(tup);
+
+    if ((dims == 0 && axis == 1) ||
+        (dims == 2 && input_shape[0] > 1 && axis == 1) ||
+        (input_shape[0] > 1 && axis == 0)) // skip since not valid case
+        return;
+
+    LayerParams lp;
+    lp.type = "Softmax";
+    lp.name = "softmaxLayer";
+    lp.set("axis", axis);
+    Ptr<SoftmaxLayer> layer = SoftmaxLayer::create(lp);
+
+    Mat input = Mat(dims, input_shape.data(), CV_32F);
+    randn(input, 0.0, 1.0);
+
+    Mat output_ref;
+    cv::exp(input, output_ref);
+    if (axis == 1){
+        cv::divide(output_ref, cv::sum(output_ref), output_ref);
+    } else {
+        cv::divide(output_ref, output_ref, output_ref);
+    }
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Softmax_Test, Combine(
+/*input blob shape*/  testing::Values(
+    make_tuple(0, std::vector<int>({0})),
```

Comment:
How 0d softmax works? How 1d tensor works with axis=1 in softmax?

---

Also I would suggest using shape vector is enough and dim can be retrieve from the vector shape.